src/platform_impl/macos/event.rs

```diff
@@ -116,7 +116,16 @@ pub(crate) fn create_key_event(
     let text_with_all_modifiers: Option<SmolStr> = if key_override.is_some() {
         None
     } else {
-        let characters = unsafe { ns_event.characters() }
+        // The logical key should always be the key resulting from applying all modifiers.
+        // Consider these cases:
+        // 1) Pressing the A key: logical key should be "a"
+        // 2) Pressing SHIFT A: logical key should be "A"
+        // 3) Pressing CTRL SHIFT A: logical key should also be "A"
+        // `ns_event.characters()` gets 3) wrong, returning "a" instead of "A".
+        // In fact it gets it wrong anytime CTRL or CMD is pressed.
+        // `ns_event.charactersIgnoringModifiers()` gets everything right.
+        // See https://github.com/rust-windowing/winit/issues/3078
+        let characters = unsafe { ns_event.charactersIgnoringModifiers() }
```

Comment:
But this doesn't make any sense though? Shift is a modifier, character ignoring modifier is just `a`, if it returns `A` it means that it accounts for shift, thus the function doesn't ignore the modifier?

This also changes the logic how `Alt+e` works and now it's always like with `OptionAsAlt`, I think. Maybe it's not bad, but it's a breaking change in behavior.

Reply:
I agree that the name `charactersIgnoringModifiers` makes no sense for how it works (I didn't name it!), I'm just reporting how it actually works on macOS Sonoma

Reply:
Hm, but Apple docs claim that they should remove the mods, like what I said is from their docs, mostly, and I know that the `Alt` behavior will change a lot, though maybe for better.

Reply:
Yeah, but we have a `key_without_modifiers` just for that... Though, the key_without_modifier is a weird in a sense that it removes the iso shift levels on linux, so you have to use the logical key stuff.

Also, what you get in `key_without_modifier` on macOS when you press `cmd + shift`?

Reply:
`charactersIgnoringModifiers` ignores all modifiers _except for shift_: https://developer.apple.com/documentation/appkit/nsevent/1524605-charactersignoringmodifiers

Reply:
…but reading the rest of the code in `create_key_event` I see a lot of convoluted logic that I don't quite follow